Xamarin.Forms.Platform.iOS/DisposeHelpers.cs

```diff
@@ -25,20 +25,25 @@ internal static void DisposeModalAndChildRenderers(this Element view)
 				}
 			}
 
-			if (view is VisualElement visualElement)
+			if (view is VisualElement visualElement
+#if __MOBILE__
+			  )
+#else
+			    && !(view is NavigationPage))
+#endif
 			{
 				renderer = Platform.GetRenderer(visualElement);
 				if (renderer != null)
 				{
+#if __MOBILE__
```

Comment:
I figured this could already be placed here since the if statement does not have content if __MOBILE__ is false.